n3fit/src/evolven3fit_new/eko_utils.py

```diff
@@ -58,26 +92,12 @@ def construct_eko_for_fit(theory_card, op_card, save_path=None):
         op_card: dict
             operator card to use for the eko
         save_path: pathlib.Path
-            path where the eko will be saved (the eko
-            won't be saved if save_path is None)
-    Returns
-    -------
-        : eko.output.Output
-        eko operator
+            path where the eko will be saved
     """
     # generate eko operator
-    if save_path is not None:
-        if not save_path.parent.exists():
-            raise FileNotFoundError(
-                f"Path where eko should be dumped does not exist: {save_path}"
-            )
-    eko_op = run_dglap(theory_card, op_card)
-    if save_path is not None:
-        # Here we want to catch all possible exceptions in order to avoid losing the computed eko
-        try:
-            _logger.info(f"Saving computed eko to : {save_path}")
-            eko_op.dump_tar(save_path)
-        except:
-            _logger.error(f"Error saving the eko to : {save_path}")
-            pass
-    return eko_op
+    if not save_path.parent.exists():
+        raise FileNotFoundError(
+            f"Path where eko should be dumped does not exist: {save_path}"
+        )
+    
+    runner.solve(theory_card, op_card, save_path)
```

Comment:
I don't think this is worth keeping a separate function for

Reply:
Also, Python philosophy is to pretend everything works fine, as long as something won't break, to save repeated checks.
In this case, EKO will try to bootstrap a new folder at the given path, and as soon as it will try to write (i.e. almost immediately) it will crash.

If you want to manage that error you can catch it with a `try... except:` (and it will be exactly a `FileNotFoundError`, returned by a `pathlib.Path.write_text()` call). But if you just raise, keep the original one, and reduce the amount of code :)

Reply:
Yes I agree, this check was done for the previous version of `eko` that computed the operator and dumped it just at the end.

Reply:
So I would remove the check but I would keep the function, even if it contains just one line, because it is used in 2 places and I believe that in this way is a bit clearer